### ISSUE TITLE
[Bug] Dropped accesories on death - Incompatiblity with Revive

### ISSUE THREAD

**Issue by salva09** [wisp-forest/accessories#78]
## Info
Minecraft version: 1.20.1
Mod loader: Fabric
Accessories version: 1.0.0-beta.26
Accessories trinkets compat version: 3.8.5-beta.5
OS: Fedora Linux 40
Java runtime: OpenJDK 17.0.11
Minecraft launcher: Prism Launcher 8.4

## Description
When playing with the accessories mod and gravestone mods, some accessories don´t get into the gravestone, and instead are dropped in place like vanilla minecraft.
This happenes for me with most accessories except for those in the spell book slot from the spell engine, bit happened with other accessoires like jewelry and backpacks

## Mods tested
Gravestones:
* Universal graves
* You're in grave danger

Accessories:
* Sophisticated backpacks
* Inmis
* Jewelry (RPG Series)

## Screenshots
![image](https://github.com/user-attachments/assets/d33ce613-8dc1-4e52-ab7c-571df1b04fd9)

<details>
  <summary>Logs</summary>

```
[21:05:53] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:54] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:54] [Render thread/ERROR]: @ Render
[21:05:54] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:54] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:55] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:55] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:56] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:56] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:57] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:57] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:58] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Render thread/ERROR]: ########## GL ERROR ##########
[21:05:58] [Render thread/ERROR]: @ Render
[21:05:58] [Render thread/ERROR]: 65539: Invalid key -1
[21:05:58] [Server thread/INFO]: sa1va01 was killed
[21:05:58] [Server thread/INFO]: [sa1va01: Killed sa1va01]
[21:05:58] [Render thread/INFO]: [System] [CHAT] sa1va01 was killed
[21:05:58] [Render thread/INFO]: [System] [CHAT] Killed sa1va01
[21:05:58] [Render thread/INFO]: Loaded 165 advancements
[21:05:58] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:59] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:05:59] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:00] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:00] [Render thread/INFO]: on Client player respawn
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/ERROR]: ########## GL ERROR ##########
[21:06:00] [Render thread/ERROR]: @ Pre render
[21:06:00] [Render thread/ERROR]: 65539: Invalid key -1
[21:06:00] [Render thread/INFO]: [System] [CHAT] Created grave at 16, 125, 3 (Overworld). It will expire in 30m
[21:06:00] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:01] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:01] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:02] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:02] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:03] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:03] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:04] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:04] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:05] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:05] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:06] [Render thread/ERROR]: Signature is missing from Property textures
[21:06:06] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:06] [Server thread/INFO]: Saving and pausing game...
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[21:06:06] [Server thread/INFO]: [STDOUT]: []
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecells:ramparts
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecells:insufferable_crypt
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/deeperdarker:otherside
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecells:black_bridge
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/spellbladenext:glassocean
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/voidz:void
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecells:promenade
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecells:prison
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/graveyard:past
[21:06:06] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/the_bumblezone:the_bumblezone
[21:06:06] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:07] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
[21:06:07] [pool-4-thread-1/WARN]: Configuration file /var/home/salva/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/Test/.minecraft/config/itemborders-common.toml is not correct. Correcting
Clipboard copy at: 05 Aug 2024 21:06:07 -0600

```

</details>

**Comment by Dragon-Seeker** [comment]
Odd question but did you have any item within the cosmetic slots of Accessories?

**Comment by salva09** [comment]
Nope, I did´t have any cosmetics on

**Comment by Dragon-Seeker** [comment]
Within testing with the latest release of accessories and the compat layer everything but the cosmetic accessories are put within the grave.

**Comment by salva09** [comment]
This is how it looks in game
![image](https://github.com/user-attachments/assets/fc7eb10b-87bd-42d8-829d-7e6a56449814)

**Comment by Dragon-Seeker** [comment]
Well, strangely such is not dropping within the prod with the example mods but could come down to some weird incompat with Universal graves.

**Comment by Dragon-Seeker** [comment]
Submit a pr to backport compat for Universal graves that may get pulled in hopefully fixing the issue fully.

**Comment by salva09** [comment]
Thank you, hopefully it will have a more predictable behaviour, because as of right now, some accessories go into the grave and others don´t

**Comment by salva09** [comment]
After updating to the latest universal grave version, 3.0.2, keeps happening
I just tested using a bare bones modpack with only the accessories, universal grave and inmis mods, and the problem didn't happen, so it probably is an incompatibility with other mod in my original modpack.
I will try and find the culprit

**Comment by Dragon-Seeker** [comment]
@salva09 You could give You're in Grave Danger a try as latest 1.20.1 release added Accessories support: https://www.curseforge.com/minecraft/mc-mods/youre-in-grave-danger/files/5606122

**Comment by salva09** [comment]
I just found out the mod to blame is Revive by GobloxZ

**Comment by Dragon-Seeker** [comment]
Honestly seems more like an incompatibility on their side. Currently no public API to resolve any problems and would recommend making a issue report on there side about this.

**Comment by phoenix492** [comment]
Has nothing to do with Revive nor Fabric. I can reproduce the issue on Forge 47.3.0 on 1.20.1 with the only mods+their dependencies being:

Accessories 1.0.0-beta26 + Curios Compat Layer 7.4.0
CERBON's Better Totem of Undying (for an accessory to test with)
and both Corpse and Gravestones individually.

(It also happens with Quark's Totem of Holding but I didn't test that individually with other mods, only in my large modpack, so I don't include it for sake of integrity of the testing)

The above mods result in accessories being flung out of the inventory upon death and never put into the gravestone/corpse. It looks like this is the exact same issue so i left a comment here, but will open a full issue report if requested.

**Comment by Dragon-Seeker** [comment]
> Has nothing to do with Revive nor Fabric. I can reproduce the issue on Forge 47.3.0 on 1.20.1 with the only mods+their dependencies being:

Such would be a different issue as it is a different set of mods altogether. Furthermore, this [issue](https://github.com/henkelmax/gravestone/issues/119) on Gravestones indicates that it will not support Curios when handling its inventory meaning it won't handle Accessories.  A similar problem with Corpse as this [issue](https://github.com/henkelmax/corpse/issues/133) also indicates the same thing.

**Comment by Dragon-Seeker** [comment]
An issue should be made on Revive to either add support for Accessories or add API to do such but I would instead recommend just using supported grave mods like [YIGD](https://modrinth.com/mod/yigd) or [Universal Graves](https://modrinth.com/mod/universal-graves)

**Comment by Dragon-Seeker** [comment]
Update on this: 53f4f9f273a748f04cfbae5e17b51f151656f7f9 should fix incompatibility for mods that don't add direct support for accessories like the ones mentioned above.

**Comment by salva09** [comment]
Cool, I'll test this out when it becomes available, thanks!

**Comment by Dragon-Seeker** [comment]
Fixed within 1.21 1.1.0 version for both before and fabric
